File: 02_Programming_Fundamentals/04_Programming_Fundamentals_PHP/17_Associative_Arrays_Exercise/05_softuni_parking.php
<?php

$n = readline();
$input = "";

$parkingRegistry = [];

for ($i = 0; $i < $n; $i++) {
  $input = explode(" ", readline());
  
  if ($input[0] == "register") {
    $username = $input[1];
    $licensePlateNumber = $input[2];

    if (!key_exists($username, $parkingRegistry)) {
      $parkingRegistry[$username] = $licensePlateNumber;
      echo "$username registered $licensePlateNumber successfully" . PHP_EOL;
    } else {
      echo "ERROR: already registered with plate number " . $parkingRegistry[$username] . PHP_EOL;
    }

  } else if ($input[0] == "unregister") {
    $username = $input[1];

    if (key_exists($username, $parkingRegistry)) {
      unset($parkingRegistry[$username]);
      echo "$username unregistered successfully" . PHP_EOL;
    } else {
      echo "ERROR: user $username not found" . PHP_EOL;
    }
  }

}

foreach ($parkingRegistry as $user => $license) {
  echo $user . " => " . $license . PHP_EOL;
}

?>

File: 02_Programming_Fundamentals/02_Programming_Fundamentals_with_C#/20_Associative_Arrays_Exercise/05_SoftUniParking.cs
using System;
using System.Linq;
using System.Collections.Generic;

class MainClass
{
    public static void Main(string[] args)
    {
        var dictParking = new Dictionary<string, string>();

        int n = int.Parse(Console.ReadLine());
        for (int i = 0; i < n; i++)
        {
            string[] input = Console.ReadLine().Split(" ");
            string operation = input[0].ToLower();

            if (operation == "register") {
                string username = input[1];
                string licensePlateNumber = input[2];

                if (!dictParking.ContainsKey(username)) {
                    dictParking[username] = licensePlateNumber;
                    Console.WriteLine($"{username} registered {licensePlateNumber} successfully");

                } else {
                    Console.WriteLine($"ERROR: already registered with plate number {dictParking[username]}");
                }

            } else if (operation == "unregister") {
                string username = input[1];

                if (!dictParking.ContainsKey(username)) {
                    Console.WriteLine($"ERROR: user {username} not found");

                }
                else {
                    dictParking.Remove(username);
                    Console.WriteLine($"{username} unregistered successfully");
                }

            }
        }

        foreach (var item in dictParking)
        {
            Console.WriteLine(item.Key + " => " + item.Value);
        }

    }
}